packages/wonder-blocks-popover/components/popover.js

```diff
@@ -0,0 +1,220 @@
+import type {
+    Placement,
+    PopperElementProps,
+} from "@khanacademy/wonder-blocks-tooltip";
+
+import PopoverContent from "./popover-content.js";
+import PopoverContentCore from "./popover-content-core.js";
+import PopoverContext from "./popover-context.js";
+import PopoverAnchor from "./popover-anchor.js";
+import PopoverDialog from "./popover-dialog.js";
+
+type Props = {|
+    ...AriaProps,
+
+    /**
+     * The element that triggers the popover. This element will be used to
+     * position the popover. It can be either a Node or a function using the
+     * children-as-function pattern to pass an open function for use anywhere
+     * within children. The latter provides a lot of flexibility in terms of
+     * what actions may trigger the `Popover` to launch the popover dialog.
+     */
+    children: React.Element<any> | (({open: () => void}) => React.Node),
+
+    /**
+     * The content of the popover. You can either use
+     * [PopoverContent](#PopoverContent) with one of the pre-defined variants,
+     * or include your own custom content using
+     * [PopoverContentCore](#PopoverContentCore directly.
+     *
+     * If the popover needs to close itself, the close function provided to this
+     * callback can be called to close the popover.
+     */
+    content:
+        | React.Element<typeof PopoverContent>
+        | React.Element<typeof PopoverContentCore>
+        | (({close: () => void}) =>
+              | React.Element<typeof PopoverContentCore>
+              | React.Element<typeof PopoverContent>),
+
+    /**
+     * Where the popover should try to appear in relation to the trigger element.
+     */
+    placement: Placement,
+
+    /**
+     * When enabled, user can hide the popover content by pressing the `esc` key
+     * or clicking/tapping outside of it.
+     */
+    dismissEnabled?: boolean,
+
+    /**
+     * The unique identifier to give to the popover. Provide this in cases where
+     * you want to override the default accessibility solution. This identifier
+     * will be applied to the popover content.
+     *
+     * By providing this identifier, the children that this popover anchors to
+     * will not be automatically given the
+     * [aria-describedby](https://www.w3.org/TR/wai-aria-1.1/#aria-describedby)
+     * attribute. Instead, the accessibility solution is the responsibility of
+     * the caller.
+     *
+     * If this is not provided, the
+     * [aria-describedby](https://www.w3.org/TR/wai-aria-1.1/#aria-describedby)
+     * attribute will be added to the children with a unique identifier pointing
+     * to the popover dialog.
+     */
+    id?: string,
+
+    /**
+     * Renders the popover when true, renders nothing when false.
+     *
+     * Using this prop makes the component behave as a controlled component. The
+     * parent is responsible for managing the opening/closing of the popover
+     * when using this prop.
+     */
+    opened?: boolean,
```

Comment:
I've found in a few places before that often, with optional `boolean` values, folks want to assign things that evaluate to `undefined` or other falsy values. I wonder if for ease of API use, we should type these as `?boolean` instead?

It's a philosphical thing, I guess. Not asking for change here, just sparking discussion.

Reply:
🤔  That's a good point. I personally prefer to enforce strict types for API consistency, but you're right that folks might want to do that as well. The good thing is that now we are going to enforce using `// @flow` on every file, so this could help to enforce these types when creating the instances. Pretty valid point tho.